src/Component/LayerPreview/LayerPreview.tsx

```diff
@@ -0,0 +1,255 @@
+  Alert,
+  Modal,
+  ModalProps,
+  Spin,
+  Tooltip
+} from 'antd';
+
+import {
+  GlobalOutlined
+} from '@ant-design/icons';
+
+import useSHOGunAPIClient from '../../Hooks/useSHOGunAPIClient';
+
+import MapComponent from './MapComponent/MapComponent';
+
+import './LayerPreview.less';
+
+function isWmsLayer(layer: OlLayer): layer is OlLayerImage<OlSourceImageWMS> | OlLayerTile<OlSourceTileWMS> {
+  if (layer instanceof OlLayer) {
+    const source = layer.getSource();
+    return source instanceof OlSourceImageWMS || source instanceof OlSourceTileWMS;
+  }
+  return false;
+}
+
+export interface LayerPreviewProps extends ModalProps {
+  layer: Layer;
+};
+
+export const LayerPreview: React.FC<LayerPreviewProps> = ({
+  layer,
+  ...passThroughProps
+}) => {
+  const [loading, setLoading] = useState(false);
+  const [addLayerError, setAddLayerError] = useState<string>();
+  const [extentError, setExtentError] = useState<string>();
+  const [loadLayerError, setLoadLayerError] = useState<string>();
+  const [open, setOpen] = useState(false);
+
+  const { t } = useTranslation();
+
+  const client = useSHOGunAPIClient();
+
+  const parser = useMemo(() => {
+    return new SHOGunApplicationUtil({
+      client: client
+    });
+  }, [client]);
+
+  const map = useMemo(() => {
+    return new OlMap({
+      layers: [
+        new OlLayerTile({
+          source: new OlSourceOSM()
+        })
+      ],
+      target: '',
+      view: new OlView({
+        center: [0, 0],
+        zoom: 0
+      })
+    });
+  }, []);
+
+  const addLayer = useCallback(async () => {
+    if (!layer) {
+      return;
+    }
+
+    try {
+      const olLayer = await parser.parseLayer(layer);
+      const layers = map.getAllLayers();
+      layers
+        .filter(l => l.get('shogunId') === layer.id)
+        .forEach(l => map.removeLayer(l));
+
+      const source = olLayer.getSource();
+
+      if (source instanceof OlSourceImageWMS) {
+        source.on('imageloaderror', (a) => {
+          setLoadLayerError(t('LayerPreview.loadLayerErrorMsg'));
+        });
+
+        source.on('imageloadend', (a) => {
+          setLoadLayerError(null);
+        });
+      }
+
+      if (source instanceof OlSourceTileWMS || source instanceof OlSourceWMTS) {
+        source.on('tileloaderror', (a) => {
+          setLoadLayerError(t('LayerPreview.loadLayerErrorMsg'));
+        });
+
+        source.on('tileloadend', (a) => {
+          setLoadLayerError(null);
+        });
+      }
+
+      if (source instanceof OlSourceVector) {
+        source.on('featuresloaderror', (a) => {
+          setLoadLayerError(t('LayerPreview.loadLayerErrorMsg'));
+        });
+
+        source.on('featuresloadend', (a) => {
+          setLoadLayerError(null);
+        });
+      }
+
+      map.addLayer(olLayer);
+    } catch (e) {
+      Logger.error(e);
+      setAddLayerError(t('LayerPreview.addLayerErrorMsg'));
+    }
+  }, [map, layer, parser, t]);
+
+  const zoomToLayer = useCallback(async () => {
```

Comment:
This function could be moved to an util or something.

Reply:
Same here.